Automation/create_vm_script.ps1

```diff
@@ -0,0 +1,98 @@
+#This script creates a Ubuntu VM on Azure.
+#The script logs user into Azure using Service Principal. To log in,
+#Tenant ID, App ID, Secret, and Subscription ID are required.
+#User must provide Azure Resource Group Name, VM Name, Admin Username/Password
+
+param (
+    [string] [Parameter(Mandatory=$true)] $tenantId, #Tenant ID used to login to Azure with Service Principal
+    [string] [Parameter(Mandatory=$true)] $applicationId, #App ID used to login to Azure with Service Principal
+    [string] [Parameter(Mandatory=$true)] $secret, #Azure Secret used to login to Azure with Service Principal
+    [string] [Parameter(Mandatory=$true)] $subscriptionId, #Subscr. ID used to login to Azure with Service Principal
+    [string] [Parameter(Mandatory)] $rgName, #Resource Group Name required
+    [string] [Parameter(Mandatory)] $vmName, #VM Name required
+    [string] [Parameter(Mandatory)] $adminUsername, #Admin Username required
+    [String] [Parameter(Mandatory)] $adminPassword, #Admin Password required
+    [string] $location='West US2', #Default value for RG/VM Location
+    [string] $vNetName='myVnet',  #Default value for Virtual Network Name
+    [string] $nsgName='myNetworkSecurityGroup', #Default value for Network Security Group Name
+    [string] $publicIpName='myPublicIpAddress' #Default value for Public IP Address Name
+    )
+
+#convert password to secure string
+[securestring]$securePassword = ConvertTo-SecureString $adminPassword -AsPlainText -Force  
+
+#template file for Virtual Machine ARM Template    
+$templateFile = "$PSScriptRoot\ubuntu_vm_parameters.json" 
+
+#Log user into Azure:
+# If the Subscription Id is NOT null, the user is logged in
+$loggedIn = ((Get-AzContext).Tenant.Id -eq $tenantId)
```

Comment:
I think Toddy is happy with the Login module, so you can replace all of this with
`Import-Module .\Login`
`Login $tenantId $applicationId $secret $subscriptionId`

(if you move this to a vm folder like I suggested, change to ..\Login)